codegen/helper.go

```diff
@@ -0,0 +1,159 @@
+package codegen
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/iancoleman/strcase"
+	"github.com/streamingfast/eth-go"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strconv"
+)
+
+type CodegenEvent struct {
+	RustName string
+	Fields   map[string]string
+}
+
+func GetContractAbi(contract string) ([]byte, error) {
+	res, err := http.Get(fmt.Sprintf("https://api.etherscan.io/api?module=contract&action=getabi&address=%s&apikey=7E11P1IJ4ZRWQ36CZN78QZ7KTE7YP7MJ31", contract))
+	if err != nil {
+		return nil, fmt.Errorf("getting contract abi from etherscan: %w", err)
+	}
+	defer res.Body.Close()
+
+	abi, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("reading fetched abi: %w", err)
+	}
+
+	type Response struct {
+		status  string
+		message string
+		Result  interface{} `json:"result"`
```

Comment:
Why it's not `string` directly since lower you do `[]byte(response.Result.(string))` which would panic if `interface{}` is not a string?

Reply:
If we keep it `interface{}`, we should check the type before accessing it and present an error message instead of a panicking.